File: app/src/main/java/fiix/challenge/fiixexercise/javasample/Listners/DataSource.java
package fiix.challenge.fiixexercise.javasample.Listners;

import android.view.View;



import java.util.List;

import fiix.challenge.fiixexercise.javasample.Model.TriviaQuestion;

public interface DataSource {
    List<TriviaQuestion> getData();
void go_to_detailpage(View view,String question, String answer,String index);
}

File: app/src/main/java/fiix/challenge/fiixexercise/kotlinsample/framework/datasource/cache/abstraction/DataSource.kt
package fiix.challenge.fiixexercise.kotlinsample.framework.datasource.cache.abstraction

import fiix.challenge.fiixexercise.kotlinsample.buisness.domain.TriviaQuestion

interface DataSource {
        fun getData(): List<String>
        fun getQuestion(): List<TriviaQuestion>
}